core/autocrypt/crypto.py

```diff
@@ -0,0 +1,378 @@
+
+    :param key: key (either public or private)
+    :type key: PGPKey
+    :return: key bytes
+    :rtype: string
+
+    """
+    assert isinstance(pgpykey, PGPKey)
+    if sys.version_info >= (3, 0):
+        keybytes = bytes(pgpykey)
+    else:
+        keybytes = pgpykey.__bytes__()
+    return keybytes
+
+
+class Crypto(object):
+    """OpenPGP operations for Autocrypt using PGPy.
+
+    PGPy does not currently support file system keyring, therefore
+    keys must be imported/exported from/to files or GPG keyring.
+
+    .. todo::
+         * instead of storing the keys in files, they could be stored
+           in the autocrypt config.json file.
+         * it probably would be better to create an abstract class
+           as interface from which both BinGPG and Crypto inherit
+           and then implement specific methods for BinGPG and Crypto.
+
+    .. note::
+         * methods not shared with BinGPG API are named as private.
+         * some operations are kept for compatibility with BinGPG API,
+           but they do not really make sense with PGPy.
+    """
+
+    def __init__(self, homedir=None, gpgpath="gpg"):
+        """Init Crypto class.
+
+        :param homedir: home dir
+        :type key: str
+        :param gpgpath: kept for compatibility with BinGPG API
+        :type gpgpath: str
+
+        """
+        # NOTE: called as .._pgpy.. to know that is instance of PGPKey
+        self.pgpydir = homedir
+        self.memkr = PGPKeyring()
+        self._ensure_init()
+
+    def __str__(self):
+        return "Crypto(homedir={homedir!r})".format(
+            homedir=self.pgpydir)
+
+    def _ensure_init(self):
+        if self.pgpydir is None:
+            return
+        if not os.path.exists(self.pgpydir):
+            os.mkdir(self.pgpydir)
+            os.chmod(self.pgpydir, 0o700)
+        self._loadkr()
+
+    def _loadkr(self):
+        keyfiles = glob.glob(os.path.join(self.pgpydir, '*.asc'))
+        self.memkr.load(keyfiles)
+
+    def _savekr(self):
+        # NOTE: saving secret keys in clear in the filesystem is a secuirty
+        # risk, but the generated keys with GPG are not passphrase protected
+        # and PGPY does not implement yet filesystem key ring
+        for fp in self.memkr.fingerprints():
+            with self.memkr.key(fp) as key:
+                self._save_key_to_file(key)
+
+    def _key_path(self, key):
+        ext = ''
+        if key.is_public is False:
+            ext = '.sec'
+        ext += '.asc'
+        keypath = os.path.join(self.pgpydir, key.fingerprint.keyid + ext)
+        return keypath
+
+    def _load_key_into_kr(self, key):
+        keypath = self._save_key_to_file(key)
+        self.memkr.load(keypath)
+
+    def _save_key_to_file(self, key):
+        keypath = self._key_path(key)
+        with open(keypath, 'wb') as fd:
+            fd.write(key_bytes(key))
+        return keypath
+
+    def _gen_skey_usage_all(self, emailadr='alice@testsuite.autocrypt.org'):
```

Comment:
a default test address is not needed in the API -- the tests specify one so i'd just keep the emailadr as a mandatory parameter.